test/new-e2e/pkg/utils/infra/stack_manager.go

```diff
@@ -142,15 +149,29 @@ func (sm *StackManager) GetStack(ctx context.Context, name string, config runner
 	return stack, upResult, err
 }
 
+// GetStackArgs are the arguments for GetStack function
+type GetStackArgs struct {
+	Context            context.Context
+	Name               string
+	Config             runner.ConfigMap
+	DeployFunc         pulumi.RunFunc
+	FailOnMissing      bool
+	LogWriter          io.Writer
+	DatadogEventSender datadogEventSender
+	UpTimeout          time.Duration
+	DestroyTimeout     time.Duration
+	CancelTimeout      time.Duration
+}
+
 // GetStackNoDeleteOnFailure creates or return a stack based on stack name and config, if error occurs during stack creation, it will not destroy the created resources. Using this can lead to resource leaks.
-func (sm *StackManager) GetStackNoDeleteOnFailure(ctx context.Context, name string, config runner.ConfigMap, deployFunc pulumi.RunFunc, failOnMissing bool, logWriter io.Writer, datadodatadogEventSender datadogEventSender) (_ *auto.Stack, _ auto.UpResult, err error) {
+func (sm *StackManager) GetStackNoDeleteOnFailure(args GetStackArgs) (_ *auto.Stack, _ auto.UpResult, err error) {
```

Comment:
Should we check that `args` contain at least some mandatory fields? Otherwise we could call `getStack` with some `nil` fields and we would get `nil pointer` panic in the code?

Reply:
Agreed!

Reply:
To make it more explicit, I will keep mandatory args in the signature and keep only optionals in args. Considering moving to functional options to follow the common pattern

Reply:
Done!

Reply:
Thanks! I think that's nice like that!